docs/pages/application-access/enroll-kubernetes-applications/architecture.mdx

```diff
@@ -0,0 +1,52 @@
+---
+title: Kubernetes Application Discovery Architecture
+description: Learn how Teleport automatically discovers applications running on Kubernetes.
+---
+
+Kubernetes application auto-discovery consists of two parts:
+
+- Periodically polling list of services in a Kubernetes cluster
+- Creating Teleport apps based on that list and proxying requests to them.
+
```

Comment:
We should add a note saying: This will only work when the teleport agent runs inside the target kubernetes cluster